cmd/juju/commands/bootstrap_test.go

```diff
@@ -314,12 +318,6 @@ var bootstrapTests = []bootstrapTest{{
 	args:     []string{"--build-agent", "--constraints", "arch=ppc64el"},
 	err:      `failed to bootstrap model: cannot use agent built for "ppc64el" using a machine running on "amd64"`,
 }, {
-	info:     "--build-agent rejects non-supported arch",
```

Comment:
IIANM, this is taken away because you changed the dummy provider. I think the dummy provider should continue rejecting some architectures, specifically so we can test things like this.